server/controllers/tasks/lib/build_task_docs.coffee

```diff
@@ -2,49 +2,82 @@ __ = require('config').universalPath
 
 module.exports = (entity)->
   Promise.all [
     searchEntityDuplicatesSuggestions entity
     getAuthorWorksData entity._id
   ]
-  .spread (suggestionEntities, authorWorksData)->
-    relationScore = calculateRelationScore suggestionEntities
-    Promise.all suggestionEntities.map(create(authorWorksData, relationScore))
-
-create = (authorWorksData, relationScore)->
-  { labels, langs } = authorWorksData
-  return (suggestionEntity)->
-    suggestionUri = suggestionEntity.uri
-    _.type suggestionUri, 'string'
-    hasWorksLabelsOccurrence suggestionUri, labels, langs
-    .then (hasOccurence)->
+  .spread (suggestions, authorWorksData)->
+    unless _.some suggestions then return []
+    Promise.all suggestions.map getOccurences authorWorksData
+    .then turnSuggestionIntoRedirection(suggestions, authorWorksData)
+    .then (occurences)->
+      Promise.all filterSuggestions occurences, suggestions
+      .then createTasksDocs(authorWorksData, occurences)
+
+turnSuggestionIntoRedirection = (suggestions, authorWorksData)->
+  return (occurences) ->
+    unless _.some(_.flattenDeep(occurences)) then return occurences
+    { labels, authorId } = authorWorksData
+    # Todo : check every labels to turn entity into redirection
+    unless canBeRedirected suggestions, labels[0] then return occurences
+    # assume first occurence is the right one to merge into
+    # only one suggestion necessary to merge
+    # occurences of first suggestion picked
+    turnIntoRedirection reconcilerUserId, authorId, occurences[0].uri
+    []
+
+filterSuggestions = (occurences, suggestions)->
+  # create a task for every suggestions
+  unless _.some(_.flattenDeep(occurences)) then return suggestions
+  # create tasks only for suggestions with occurences
+  suggestions.filter suggestionsWithOccurences occurences
+
+createTasksDocs = (authorWorksData, occurences) ->
+  return (suggestions) ->
+    relationScore = calculateRelationScore suggestions
+    return suggestions.map create authorWorksData, relationScore, occurences
+
+canBeRedirected = (suggestions, workLabel) ->
+  # several suggestions == has homonym
+  unless suggestions.length == 1 then return false
+  workLabel.length > 12
+
+suggestionsWithOccurences = (occurences)->
+  return (suggestions)->
+    suggestionsUris = _.pluck suggestions, 'uri'
+    occurencesUris = _.pluck occurences, 'uri'
+    _.intersection suggestionsUris, occurencesUris
+
+getOccurences = (authorWorksData)->
+  return (suggestion)->
+    { labels, langs } = authorWorksData
+    hasWorksLabelsOccurrence suggestion.uri, labels, langs
+    .then (occurences)->
+      if _.isEmpty(_.flattenDeep(occurences)) then return []
+      return
+        uri: suggestion.uri
+        occurences: occurences
+
+create = (authorWorksData, relationScore, suggestionsOccurences)->
+  return (suggestion)->
+    { authorId } = authorWorksData
+    suggestionOccurences = _.filter(suggestionsOccurences, {'uri': suggestion.uri})
```

Comment:
this means looping through `suggestionsOccurences.length` times over.
A work around would be to create an index in the higher order function:
```coffee
# between L70-L71
occurrencesBySuggestionUri = _.indexBy suggestionsOccurences, 'uri'
```
```coffee
suggestionOccurences = occurrencesBySuggestionUri[suggestion.uri]
```
(unless there are several `suggestionsOccurences` with the same uri, in which case, why taking only the first one into account here after?)